Automattic/wp-calypso
# NUX: Always submit the original English site topic if it's from the suggestions

#### Changes proposed in this Pull Request

This PR adds the code that maps the translated site topic back to its original English slug. It also fixes the bug that the topic suggestions list isn't always localized.

The reason that the topic suggestions list isn't always localized currently is because `lib/hint-data` defines it as a module-level variable. Thus, any locale update simply won't update it and it will just stay in English. Moving it to the component level makes sure it gets updated properly.

A screencast of this PR in action:
![demo-site-topic-slug](https://user-images.githubusercontent.com/1842898/49426735-5ffbbf80-f7dc-11e8-8e3e-4bffcef231e9.gif)


#### Testing instructions

1. Change your locale to something other than English.
1. Go through the improved onboarding flow from http://calypso.localhost:3000/start/onboarding
1. At the site topic step, use auto-completion to pick a topic.
1. After that, check `getState().signup` and make sure it has been mapped back to the original English slug.
1. Redo 1~2. Then, at the site topic step, enter an arbitrary string.
1. Check `getState().signup` and it should be whatever you just have entered.
1. Go through the current main flow from http://calypso.localhost:3000/start/main, and the site topic field from the About step should act the same as above.
